wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession+WireCallStateObserver.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAnalytics
+import WireDataModel
+
+extension ZMUserSession: WireCallCenterCallStateObserver {
+    public func callCenterDidChange(
+        callState: CallState,
+        conversation: ZMConversation,
+        caller: any UserType,
+        timestamp: Date?,
+        previousCallState: CallState?
+    ) {
+
+        switch (callState, previousCallState) {
+        case (.outgoing, _):
+            let isVideoCall = isVideoCall(for: conversation)
+            trackCallInitialized(isVideo: isVideoCall, conversationType: conversation.conversationType)
+        case (.answered(_), .incoming(video: let isVideoCall, shouldRing: _, degraded: _)):
```

Comment:
Neither of these call states represent the self user actually being in the call. I think you need `.established`.

Reply:
I misread the switch statement, I didn't realize you're comparing previous and current call states. This code assumes that `incoming` always precedes `answered`, which might not be guaranteed.

What is the intention of comparing call states? To avoid repeated event tracking for a single call?

Reply:
We thought it as a solution with @netbe so we can get the value for `isVideoCall` for the call joined event. Otherwise the value would be set to always false.

Reply:
Can you read of from the voice channel, just like the outgoing case?

Reply:
so actually yes you can but it only covers the oneonone. I retested and if we do a group call the isVideo will be false all the time, the info should be available in the call participants though

Reply:
After discussing with @netbe we reverted it to the original implementation.